.gitattributes

```diff
@@ -0,0 +1,2 @@
+# treat patches as files that should not be modified
+*.patch -text
```

Comment:
can we just do `* -text` to prevent git from ever munging files? (similar to what we suggest at https://github.com/JuliaLang/julia/blob/master/README.windows.md#line-endings)?

Reply:
isn't that different? we suggest `eol=lf` and `autocrlf=input`, while `-text` implies no `autocrlf` at all (potentially introducing crlf line endings)